osbuild/testutil/__init__.py

```diff
@@ -149,7 +157,7 @@ def make_container(tmp_path, fake_content, base="scratch"):
     try:
         yield container_id
     finally:
-        subprocess.check_call(["podman", "image", "rm", container_id])
+        subprocess.check_call(["podman", "image", "rm", "--no-prune", container_id])
```

Comment:
@mvo5 / @achilleas-k any good idea why this worked on your end?
when creating layers with this nested "with make_container()" construct I got an error when destructing the outer one...

Reply:
Thank you, this is an excellent find. I ran the other test that uses this nesting and it is failing without the change so something is fishy. I opened https://github.com/osbuild/osbuild/pull/1755 to see why the relevant test(s) are skipped in GH.

Reply:
A comment here is probably also needed and I would put it into it's own commit for posterity (i.e. it will be easier to see what is going on in git log/git blame). And I wonder if `--ignore` or `--no-prune` is the better option here.

Reply:
`--ignore` would probably not cover the nested usecase correctly as this would remove the upper layer if you just leave the lower one - where you might be still using it